addons/delivery/models/stock_picking.py

```diff
@@ -103,6 +106,14 @@ def _compute_return_picking(self):
             else:
                 picking.is_return_picking = False
 
+    @api.depends('state', 'carrier_tracking_ref', 'carrier_id', 'weight_at_label', 'shipping_weight', 'picking_type_id')
+    def _compute_show_print_label_button(self):
+        for pick in self:
+            if pick.picking_type_id.print_label and ((pick.state == 'assigned' and not pick.carrier_tracking_ref and pick.carrier_id) or (pick.state == 'done' and pick.weight_at_label > 0 and pick.weight_at_label != pick.shipping_weight)):
```

Comment:
Maybe divide this condition on multiple lines :sweat_smile: 
```suggestion
            if pick.picking_type_id.print_label and (
               (pick.state == 'assigned' and not pick.carrier_tracking_ref and pick.carrier_id)
               or (pick.state == 'done' and pick.weight_at_label > 0 and pick.weight_at_label != pick.shipping_weight)):
```